src/components/edit/editLineForm/MakbilaList.tsx

```diff
@@ -33,77 +35,231 @@ export const MakbilaMenu = (props: Props) => {
+          tractate,
+          chapter,
+          mishna: currentMishna.mishna,
+          lineNumber: currentLineNumber,
+          selectedSublineIndices: link.currentSublineIndices,
+          currentSublineIndices: link.selectedSublineIndices,
+        };
+        
+        await LineService.addParallel(
+          link.tractate,
+          link.chapter || '',
+          link.mishna || '',
+          link.lineNumber || '',
+          reciprocalParallel
+        );
+      } else {
+        // For cross-mishna parallels, single operation (backend handles reciprocal)
+        await LineService.addParallel(tractate, chapter, currentMishna.mishna, currentLineNumber, link);
+      }
       
-      // Success notification
-      showSuccess(successMessage);
+      // Refresh the UI by refetching the mishna
+      const updatedMishna = await PageService.getMishna(tractate, chapter, currentMishna.mishna);
+      const currentLine = updatedMishna.lines?.find(line => line.lineNumber === currentLineNumber);
+      // PageService.getMishna() already converts sublinePairs to selectedSublineIndices
+      const updatedParallels = currentLine?.parallels || [];
       
+      onUpdateInternalSources(updatedParallels);
+      showSuccess('המקבילה נוספה בהצלחה!');
     } catch (error) {
-      // Error notification
-      showError(`${errorPrefix}: ${error instanceof Error ? error.message : 'Unknown error'}`);
+      console.error('Error adding parallel:', error);
+      showError('שגיאה בהוספת המקבילה');
     }
   };
   
-  // Handle adding a new parallel
-  const handleAddParallel = async (link: iParallelLink) => {
-    const updatedParallels = [...parallels, link];
-    await saveParallelsToDb(updatedParallels, 'המקבילה נשמרה בהצלחה!', 'שגיאה בשמירת המקבילה');
-  };
-  
-  // Handle deleting a parallel
+  // Handle deleting a parallel (using NEW granular operation)
   const handleDeleteParallel = async (index: number) => {
-    const updatedParallels = [...parallels];
-    updatedParallels.splice(index, 1);
-    await saveParallelsToDb(updatedParallels, 'המקבילה נמחקה בהצלחה!', 'שגיאה במחיקת המקבילה');
+    const tractate = getTractate(currentMishna);
+    const chapter = getChapter(currentMishna);
+    
+    if (!tractate || !chapter) {
+      showError('לא ניתן למחוק - מידע המשנה לא זמין עדיין');
+      return;
+    }
+
+    const parallelToDelete = parallels[index];
+    if (!parallelToDelete) {
+      showError('מקבילה לא נמצאה');
+      return;
+    }
+
+    try {
+      // Check if it's a same-mishna parallel
+      const isSameMishna = parallelToDelete.tractate === tractate && 
+                          parallelToDelete.chapter === chapter && 
+                          parallelToDelete.mishna === currentMishna.mishna;
+      
+      if (isSameMishna) {
+        // For same-mishna parallels, delete both sides sequentially
+        await LineService.deleteParallel(tractate, chapter, currentMishna.mishna, currentLineNumber, parallelToDelete);
```

Comment:
I think the backend should treat it (think it already does) - when we delete a parallel - it should delete both sides